cmd/checkheader/main.go

```diff
@@ -104,8 +106,21 @@ func generate(header string) (string, error) {
 	var b strings.Builder
 	sc := bufio.NewScanner(f)
 	var i int
+	var enum bool
 	for sc.Scan() {
 		l := strings.TrimSpace(sc.Text())
+		if enum {
+			if strings.HasPrefix(l, "}") {
+				enum = false
+				continue
+			}
+			tryConvertEnum(&b, l)
+			continue
+		}
+		if strings.HasPrefix(l, "enum {") {
+			enum = true
+			continue
+		}
```

Comment:
_[JavaScript mode on]_
Lets pull in https://github.com/qmuntal/stateless so we can track that single-bit state using a proper state machine library.
_[JavaScript mode off]_

Nitpick done 😄

Reply:
I wasn't clear enough about my nitpick, sorry--I meant the outer `if`, not the `if !strings.HasPrefix(l, "}") {`, so:

```go
		if strings.HasPrefix(l, "enum {") {
			enum = true
			continue
		}
		if enum {
			if strings.HasPrefix(l, "}") {
				enum = false
			} else {
				tryConvertEnum(&b, l)
			}
			continue
		}
```

(Also tried flipping the inner `if` here to avoid the `!`, which I tend to like when there's no bigger reason to order the cases a particular way.)

I approve either way, but figured I'd follow up while I'm in nitpicking mode anyway. 🙂

Reply:
Now that I see it written out, I see it'd let this work when it should actually fail:

```
enum {
enum {
  GO_FOO = 2
}
```

Doesn't matter (since compilation should have failed anyway), but for the sake of correctness it seems right to not do this flip.